api/command.go

```diff
@@ -571,3 +577,110 @@ func loadTestPostsCommand(c *Context, command *model.Command) bool {
 
 	return false
 }
+
+func suggestFromWebHooks(c *Context, command *model.Command) bool {
+	if result := <-Srv.Store.User().Get(c.Session.UserId); result.Err != nil {
+		return false
+	} else {
+		var user = result.Data.(*model.User)
+
+		hchan := Srv.Store.Webhook().GetOutgoingByTeam(c.Session.TeamId)
+		tchan := Srv.Store.Team().Get(c.Session.TeamId)
+		cchan := Srv.Store.Channel().Get(command.ChannelId)
+
+		var team *model.Team
+		if result := <-tchan; result.Err != nil {
+			l4g.Error("Encountered error getting team, team_id=%s, err=%v", c.Session.TeamId, result.Err)
+			return false
+		} else {
+			team = result.Data.(*model.Team)
+		}
+
+		var channel *model.Channel
+		if result := <-cchan; result.Err != nil {
+			l4g.Error("Encountered error getting channel, channel_id=%s, err=%v", command.ChannelId, result.Err)
+			return false
+		} else {
+			channel = result.Data.(*model.Channel)
+		}
+
+		hooks := []*model.OutgoingWebhook{}
+
+		if result := <-hchan; result.Err != nil {
+			l4g.Error("Encountered error getting webhooks by team, err=%v", result.Err)
+			return false
+		} else {
+			hooks = result.Data.([]*model.OutgoingWebhook)
+		}
+
+		if len(hooks) == 0 {
+			return false
+		}
+
+		firstWord := strings.Split(command.Command, " ")[0]
+
+		relevantHooks := []*model.OutgoingWebhook{}
+
+		for _, hook := range hooks {
+			if hook.ChannelId == command.ChannelId {
+				if len(hook.TriggerWords) == 0 || hook.HasTriggerWord(firstWord) {
+					relevantHooks = append(relevantHooks, hook)
+				}
+			} else if len(hook.ChannelId) == 0 && hook.HasTriggerWord(firstWord) {
+				relevantHooks = append(relevantHooks, hook)
+			}
+		}
+
+		done := make(chan bool)
+		hooksToCall := len(relevantHooks)
+		for _, hook := range relevantHooks {
+			go func(hook *model.OutgoingWebhook) {
+				p := url.Values{}
```

Comment:
It's not really necessary,the whole function block just got a bit big and I think it's easier to read if split into multiple small functions.
The way url.Values is used in this case is more ore less like a query builder (that is put stuff into a slice/array then call encode to make it a query string like ?foo=bar&...)

If you're coming from a php background, basically what it does is somethink like this:
```php
$token = 123;
$teamId = 9;

$p = array();
$p['token'] = $token;
$p['teamId'] = $teamId;
//....
$query = http_build_query($p);

echo $query; // token=123&teamId=9
```